### PR TITLE
added channel cookies and celery task

### PR DESCRIPTION
**Pull request by Mikhail-Gorelov**

### CODE REVIEW

File: web/api/v1/products_microservice/services.py [hunk 1]
```diff
@@ -13,3 +13,32 @@ def custom_headers(self) -> dict:
...
+            return next(item for item in self.microservice_response.data.get('results')
+                        if item["country"] == self.request.data.get('country'))
+        except StopIteration:
+            return {'country': 'DE'}
+
+    def set_channel_cookie(self):
+        refresh_cookie_path = getattr(settings, 'JWT_AUTH_REFRESH_COOKIE_PATH', '/')
+        cookie_secure = getattr(settings, 'JWT_AUTH_SECURE', False)
+        cookie_httponly = getattr(settings, 'JWT_AUTH_HTTPONLY', True)
+        cookie_samesite = getattr(settings, 'JWT_AUTH_SAMESITE', 'Lax')
```
Comment (bandirom) [inline web/api/v1/products_microservice/services.py]: why do you use jwt settinhs? :)

File: web/api/v1/products_microservice/views.py [hunk 2]
```diff
@@ -97,3 +91,38 @@ def get(self, request, *args, **kwargs):
...
         return Response(response.data)
+
+
+class ChannelListView(GenericAPIView):
+    permission_classes = (AllowAny,)
+
+    def get(self, request):
+        service = ProductsService(request=request, url=f"/api/v1/channel-list/")
+        response = service.service_response(method="get")
+        return Response(response.data)
```
Comment (bandirom) [inline web/api/v1/products_microservice/views.py]: `return response`  will work as well

File: web/main/services.py [hunk 3]
```diff
@@ -8,3 +27,58 @@ class RemoteUser:
...
+        cookie_samesite = getattr(settings, 'JWT_AUTH_SAMESITE', 'Lax')
+        return self.response.set_cookie(
+            key=settings.CHANNEL_SETTINGS['COOKIE_NAME'],
+            value=current_channel,
+            max_age=settings.CHANNEL_SETTINGS['CACHE_ACTIVE_CHANNELS_TIMEOUT'],
+            secure=cookie_secure,
+            httponly=cookie_httponly,
+            samesite=cookie_samesite,
+            path=refresh_cookie_path,
+        )
```
Comment (bandirom) [inline web/main/services.py]: duplicated with  ` SetChannelCookieService.set_channel_cookie`

File: web/main/services.py [hunk 4]
```diff
@@ -8,3 +27,58 @@ class RemoteUser:
...
+        channel_settings = settings.CHANNEL_SETTINGS
+        ip_address = channel_settings.get('FORCED_IP_ADDRESS', get_remote_ip_from_request(self.request))
+        ip_server_url = channel_settings.get('GET_IP_SERVER').format(ip_address=ip_address)
+        r = requests.get(ip_server_url)
+        if not r.status_code == HTTP_200_OK:
+            logging.error(f'Ip server returned {r.status_code} {r.json()}')
+            return {'country': 'DE'}
+        return r.json()
+
+    def check_active_channels_cache_key(self) -> None:
```
Comment (bandirom) [inline web/main/services.py]: why called `check...` if you set the coockies there

File: web/main/tasks.py [hunk 5]
```diff
@@ -1,50 +1,19 @@
...
-        subject=subject,
-        from_email=kwargs.get('from_email'),
-        to=to_email,
-        bcc=kwargs.get('bcc'),
-        cc=kwargs.get('cc'),
-        reply_to=kwargs.get('reply_to'),
+@app.task
+def cache_active_channels():
+    if cache.keys(search=f'*{settings.CHANNEL_SETTINGS["CACHE_ACTIVE_CHANNELS_KEY"]}'):
+        return
```
Comment (bandirom) [inline web/main/tasks.py]: you need to force caching

File: web/main/tasks.py [hunk 6]
```diff
@@ -1,50 +1,19 @@
...
+    service = CeleryProductsService(url=f"{settings.CHANNEL_SETTINGS['ACTIVE_CHANNELS_URL']}")
```
Comment (bandirom) [inline web/main/tasks.py]: I'd to integrate your ProductService instead of creationg new one

File: web/main/tasks.py [hunk 7]
```diff
@@ -1,50 +1,19 @@
...
+    response = service.service_response(method="get")
+    cache_key = cache.make_key(settings.CHANNEL_SETTINGS["CACHE_ACTIVE_CHANNELS_KEY"], response.data.get('results'))
```
Comment (bandirom) [inline web/main/tasks.py]: why you put response data to the key?

File: web/api/v1/products_microservice/services.py (hunk 1 above)
Comment (Mikhail-Gorelov) [reply web/api/v1/products_microservice/services.py]: Agree, there is no use in this option, going to remove it in next commit

File: web/api/v1/products_microservice/views.py (hunk 2 above)
Comment (Mikhail-Gorelov) [reply web/api/v1/products_microservice/views.py]: Agree, will change that

File: web/main/services.py (hunk 3 above)
Comment (Mikhail-Gorelov) [reply web/main/services.py]: Agree

File: web/main/services.py (hunk 4 above)
Comment (Mikhail-Gorelov) [reply web/main/services.py]: Going to rename it to set_active_channels_cache_key

File: web/main/tasks.py (hunk 6 above)
Comment (Mikhail-Gorelov) [reply web/main/tasks.py]: ProductService is inherited from MicroServiceConnect. ProductService requires request, which we don't have in celery task. CeleryProductsService is inherited from ConnectionService, which does not require request. That's it :)

File: web/main/tasks.py (hunk 5 above)
Comment (Mikhail-Gorelov) [reply web/main/tasks.py]: Okay, I will remove this condition and let the task purely cache every 20 minutes. But the reasons I had is to check whether we have a cache than we could just skip caching!

File: web/main/tasks.py (hunk 7 above)
Comment (Mikhail-Gorelov) [reply web/main/tasks.py]: agree, there is no reasons for that. my fault, going to remove that!

File: web/main/tasks.py (hunk 6 above)
Comment (bandirom) [reply web/main/tasks.py]: thats why you can try to use ConnectionService for ProductService

Comment (Mikhail-Gorelov) [reply web/main/tasks.py]: But in ProductService we have custom_headers in which we set headers. Inside this method we have self.request, which we only have in MicroServiceConnect.

**Pull request merged**